Rasa_Bot/actions/actions.py

```diff
@@ -906,4 +910,70 @@ async def run(self, dispatcher, tracker, domain):
                                     definitions.TrackerName.SMOKING.value,
                                     "This is a tracker",
                                     recursive_rule)
-        return[]
+        return []
+
+
+# Store last intervention component in database
+class StoreLastInterventionComponent(Action):
+    def name(self):
+        return "action_store_intervention_component"
+
+    async def run(self, dispatcher, tracker, domain):
+        user_id = int(tracker.current_state()['sender_id'])  # retrieve userID
+
+        slot = tracker.get_slot("current_intervention_component")
+        logging.info(slot)
+
+        celery.send_task('celery_tasks.dialog_completed', (user_id, slot))
+        logging.info("no celery error")
+
+        return []
+
+
+### Slot-setting methods called for rasa to store current intervention component
+class SetSlotProfileCreation(Action):
+    def name(self):
+        return "action_slot_profile_creation"
+
+    async def run(self, dispatcher, tracker, domain):
+        return [SlotSet("current_intervention_component", PreparationDialogs.PROFILE_CREATION)]
+
+
+class SetSlotMedicationTalk(Action):
+    def name(self):
+        return "action_slot_medication_talk"
+
+    async def run(self, dispatcher, tracker, domain):
+        return [SlotSet("current_intervention_component", PreparationDialogs.MEDICATION_TALK)]
+
+
+class SetSlotColdTurkey(Action):
+    def name(self):
+        return "action_slot_cold_turkey"
+
+    async def run(self, dispatcher, tracker, domain):
+        return [SlotSet("current_intervention_component", PreparationDialogs.COLD_TURKEY)]
+
+
+class SetSlotPlanQuitStart(Action):
+    def name(self):
+        return "action_slot_plan_quit_start"
+
+    async def run(self, dispatcher, tracker, domain):
+        return [SlotSet("current_intervention_component", PreparationDialogs.PLAN_QUIT_START_DATE)]
```

Comment:
```suggestion
class SetSlotPlanQuitStartDate(Action):
    def name(self):
        return "action_slot_plan_quit_start_date"

    async def run(self, dispatcher, tracker, domain):
        return [SlotSet("current_intervention_component", PreparationDialogs.PLAN_QUIT_START_DATE)]
```
Would be good to have consistent naming